src/theory/bv/theory_bv_rewrite_rules_simplification.h

```diff
@@ -1557,6 +1557,40 @@ Node RewriteRule<ShiftZero>::apply(TNode node) {
 
 /* -------------------------------------------------------------------------- */
 
+/**
+ * UgtUrem
+ *
+ * (bvugt (bvurem T x) x)
+ *   ==>  (ite (= x 0_k) (bvugt T x) false)
+ *   ==>  (and (=> (= x 0_k) (bvugt T x)) (=> (not (= x 0_k)) false))
+ *   ==>  (and (=> (= x 0_k) (bvugt T x)) (= x 0_k))
+ *   ==>  (and (bvugt T x) (= x 0_k))
+ *   ==>  (and (bvugt T 0_k) (= x 0_k))
+ */
+
+template<> inline
+bool RewriteRule<UgtUrem>::applies(TNode node) {
+  return (
+      node.getKind() == kind::BITVECTOR_UGT &&
+      node[0].getKind() == kind::BITVECTOR_UREM_TOTAL &&
+      node[0][1] == node[1]
+  );
+}
+
+template<> inline
+Node RewriteRule<UgtUrem>::apply(TNode node) {
+  Debug("bv-rewrite") << "RewriteRule<UgtUrem>(" << node << ")" << std::endl;
+  const Node& T = node[0][0];
+  const Node& x = node[1];
+  NodeManager* nm = NodeManager::currentNM();
+  return nm->mkNode(kind::AND,
+    nm->mkNode(kind::EQUAL, x, utils::mkConst(utils::getSize(x), 0)),
```

Comment:
How about you create a separate `Node zero = utils::mkConst(utils::getSize(x), 0)` since you need it twice?

Reply:
Done.